### PR TITLE
contrib: remove outdated examples and files

### PR DESCRIPTION
**Pull request by thaJeztah**
-  contrib: remove outdated "desktop-integration" examples
  These Dockerfiles haven't been maintained, and more up-to-date versions can be found in Jess's github repository;
    - chromium: https://github.com/jessfraz/dockerfiles/tree/3679486e0b2f92fca296fcd531f34747a6af3315/chromium
    - gparted: https://github.com/jessfraz/dockerfiles/tree/3679486e0b2f92fca296fcd531f34747a6af3315/gparted
- contrib: remove outdated vagrant docs
- contrib: remove unused REVIEWERS files

### CODE REVIEW

File: contrib/desktop-integration/README.md [hunk 1]
```diff
@@ -1,11 +0,0 @@
-Desktop Integration
```
Comment (AkihiroSuda) [inline contrib/desktop-integration/README.md:1]: nit: Maybe we can keep this README and add a link to https://github.com/jessfraz/dockerfiles/tree/master

**Review by AkihiroSuda — APPROVED** [review]
LGTM, maybe we can keep README but I don't have a strong opinion.

File: contrib/desktop-integration/README.md (hunk 1 above)
Comment (thaJeztah) [reply contrib/desktop-integration/README.md:1]: Yeah, I was considering that. Not sure how many people actually discover these (I think it'd be very low), so didn't think it was worth doing.

**Comment by thaJeztah** [comment]
Found some more old stuff that's probably ok to remove; hold on, I'll push some commits

**Comment by thaJeztah** [comment]
ok, pushed two more commits. Also looking at the `gitdm` directory (which was added in https://github.com/moby/moby/pull/23085) but can do that one separately

File: contrib/init/upstart/REVIEWERS [hunk 2]
```diff
@@ -1,2 +0,0 @@
-Tianon Gravi <admwiggin@gmail.com> (@tianon)
```
Comment (AkihiroSuda) [inline contrib/init/upstart/REVIEWERS:1]: Why not remove the entire upstart stuff?

Comment (thaJeztah) [reply contrib/init/upstart/REVIEWERS:1]: TBH; not sure if people are still using them 🤔 I recall when we stopped packaging it in our packages, that some people ran on non-systemd distros, and opened PR's to bring some of the things back; https://github.com/docker/docker-ce-packaging/blob/fcbe36668d1a66a2f7679ad191c9165c37c902e2/deb/common/docker-ce.docker.init#L49 (https://github.com/docker/docker-ce-packaging/pull/287).

I can have a look at removing them (and check if there's things in these directories that we missed in the docker-ce-packaging repo). Think that's better to do in a separate PR for visibility

**Review by AkihiroSuda — APPROVED** [review]

**Review by tiborvass — APPROVED** [review]

**Pull request merged**